molgenis-data-rest/src/test/java/org/molgenis/data/rest/v2/RestControllerV2Test.java

```diff
@@ -344,6 +368,208 @@ public void retrieveResourceCollection() throws Exception
 				.andExpect(content().string(resourceCollectionResponse));
 	}
 
+	@Test
+	public void testCreateEntities() throws Exception
+	{
+		String content = "{entities:[{id:'p1', name:'Piet'}, {id:'p2', name:'Pietje'}]}";
+		String responseBody = "{\n  \"resources\": [\n    {\n      \"href\": \"/api/v2/entity/p1\"\n    },\n"
+				+ "    {\n      \"href\": \"/api/v2/entity/p2\"\n    }\n  ]\n}";
+		mockMvc.perform(post(HREF_ENTITY_COLLECTION).content(content).contentType(APPLICATION_JSON))
+				.andExpect(status().isCreated()).andExpect(content().contentType(APPLICATION_JSON))
+				.andExpect(header().string("Location", "/api/v2/entity?q=id=in=(p1,p2)"))
+				.andExpect(content().string(responseBody));
+
+		verify(dataService).add(Matchers.eq(ENTITY_NAME), Matchers.anyListOf(MapEntity.class));
+	}
+
+	/**
+	 * Test all the exception that the method 'updateEntitiesExceptions' can throw
+	 * 
+	 * @throws Exception
+	 */
+	@Test
+	public void testCreateEntitiesExceptions() throws Exception
+	{
+		this.testCreateEntitiesExceptions("entity", "{}", RestControllerV2.EXCEPTION_NO_ENTITIES);
+
+		this.testCreateEntitiesExceptions("entity", this.createMaxPlusOneEntitiesAsTestContent(),
+				RestControllerV2.EXCEPTION_MAX_ENTITIES_EXCEEDED);
+
+		this.testCreateEntitiesExceptions("entity2", "{entities:[{email:'test@email.com', extraAttribute:'test'}]}",
+				RestControllerV2.createUnknownEntityException("entity2"));
+
+		this.testCreateEntitiesExceptions("entity", "{entities:[{email:'test@email.com', extraAttribute:'test'}]}",
+				RestControllerV2.createMolgenisDataExceptionUnknownIdentifier(0));
+	}
+
+	@Test
+	public void testUpdateEntities() throws Exception
+	{
+		String content = "{entities:[{id:'p1', name:'Witte Piet'}, {id:'p2', name:'Zwarte Piet'}]}";
+		mockMvc.perform(put(HREF_ENTITY_COLLECTION).content(content).contentType(APPLICATION_JSON)).andExpect(
+				status().isOk());
+
+		verify(dataService, times(1)).update(Matchers.eq(ENTITY_NAME), Matchers.anyListOf(MapEntity.class));
+	}
+
+	/**
+	 * Test all the exception that the method 'updateEntitiesExceptions' can throw
+	 * 
+	 * @throws Exception
+	 */
+	@Test
+	public void testUpdateEntitiesExceptions() throws Exception
+	{
+		this.testUpdateEntitiesExceptions("entity", "{}", RestControllerV2.EXCEPTION_NO_ENTITIES);
```

Comment:
I really like the utility method, but I think the tests are worth separate test methods to make sure all tests get run even if one fails.